File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int len = prices.size();

        int buy = INT_MAX;
        int profit = 0;
        for (int i = 0; i < len; i++) {
            buy = min(buy, prices[i]);

            if (prices[i] > buy) {
                profit = max (profit, prices[i] - buy);
            }
        }

        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int ans = 0;

        int minTillNow = (int) 1e9;
        for (int curr : prices) {
            if (curr > minTillNow) {
                ans = Math.max(ans, curr - minTillNow);
            }

            minTillNow = Math.min(minTillNow, curr);
        }

        return ans;
    }
}